Realm/RLMSyncSession.h

```diff
@@ -150,8 +152,20 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (nullable RLMProgressNotificationToken *)addProgressNotificationForDirection:(RLMSyncProgressDirection)direction
                                                                           mode:(RLMSyncProgress)mode
+                                                           dispatchToMainQueue:(BOOL)dispatchToMainQueue
```

Comment:
What do we gain from the user explicitly specifying a queue? Why is that preferable to having them dispatch to whichever queue they care about from within the callback?